### PR TITLE
Emit the onnx-mlir version as llvm.ident metadata.

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Signed-off-by: Whitney Tsang <whitneyt@ca.ibm.com>

### CODE REVIEW

**Comment by tungld** [comment]
@jenkins-droid test this please

File: src/Compiler/CompilerUtils.cpp [hunk 1]
```diff
@@ -521,6 +521,15 @@ static void genLLVMBitcode(const mlir::OwningOpRef<ModuleOp> &module,
     llvm::errs() << "Failed to translate module to LLVMIR.\n";
     exit(1);
   }
+
+  // Emit the onnx-mlir version as llvm.ident metadata.
+  llvm::NamedMDNode *IdentMetadata =
+      llvmModule->getOrInsertNamedMetadata("llvm.ident");
+  std::string Version = "onnx-mlir version 1.0.0";
+  llvm::LLVMContext &Ctx = llvmModule->getContext();
```
Comment (etiotto) [inline src/Compiler/CompilerUtils.cpp:529]: [nit]: Ctx -> ctx

Comment (etiotto) [reply src/Compiler/CompilerUtils.cpp:529]: note that, unlike in LLVM, variable start with a lowercase character.

**Review by etiotto — COMMENTED** [review]
Possible to add a lit test ?

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by etiotto** [comment]
@whitneywhtsang seems like the new lit test failed on Win. Please take a look.

**Comment by whitneywhtsang** [comment]
@jenkins-droid test this please

**Comment by etiotto** [comment]
@jenkins-droid test this please

**Review by etiotto — APPROVED** [review]
LGTM

File: src/Compiler/CompilerUtils.cpp [hunk 2]
```diff
@@ -532,6 +532,14 @@ static void genLLVMBitcode(const mlir::OwningOpRef<ModuleOp> &module,
     }
   }
 
+  // Emit the onnx-mlir version as llvm.ident metadata.
+  llvm::NamedMDNode *identMetadata =
+      llvmModule->getOrInsertNamedMetadata("llvm.ident");
+  std::string version = "onnx-mlir version 1.0.0";
```
Comment (tungld) [inline src/Compiler/CompilerUtils.cpp:538]: According to our release notes, the latest version is 0.2.0 not 1.0.0. But given that this version is for the current code, I recommend to include git commit id which is more useful. For example, in clang, I see this
```
!llvm.ident = !{!5}
!5 = !{!"clang version 14.0.0 (git@github.ibm.com:compiler/llvm-project.git 033b8a4be1acb95d46ba510c7329977ceb80a680)"}
```

Can we have a global place for the version number? I believe we should make the version info consistent across the whole code. At least, a global variable in CompilerUtils.cpp would be helpful.

Comment (etiotto) [reply src/Compiler/CompilerUtils.cpp:538]: I spoke to @whitneywhtsang to do this in a separate PR.

Comment (whitneywhtsang) [reply src/Compiler/CompilerUtils.cpp:538]: https://github.com/onnx/onnx-mlir/pull/1215

**Pull request merged**